### DISCUSSION TITLE
How do I test that analytical gradients match numerical ones?

### DISCUSSION THREAD

**Discussion by NeilGirdhar** [jax-ml/jax#23134] [General]
I could have sworn I've seen code like this in Jax somewhere.  Is there an easy tool to check that `grad(f)(x)` is approximately equal to `(f(x)-f(x+delta))/delta` for small delta?

**Comment by NeilGirdhar** [comment]
Found it: `jax.test_util.check_grads`.
